client/state/document-head/reducer.js

```diff
@@ -18,7 +18,7 @@ import {
 export function title( state = '', action ) {
 	switch ( action.type ) {
 		case DOCUMENT_HEAD_TITLE_SET:
-			return action.title;
+			return action.title || state;
```

Comment:
@aduth: Fix in 960f9cc8c6ff56a60dc91807eff160b0b65416e6.

A condition like `if ( 'title' in this.props ) { ... }` (as originally suggested [here](https://github.com/Automattic/wp-calypso/pull/7224#r74437843)) doesn't catch `this.props.title` being undefined, which would hence be used when dispatching the action.